picker/picker.go

```diff
@@ -105,11 +105,16 @@ func (mwr *remotesWeightedRandom) Select(excludes ...string) (api.Peer, error) {
 	cum := 0.0
 	// calculate CDF over weights
 	for peer, weight := range mwr.remotes {
+		find:=false
```

Comment:
I see it now: rather than hack this loop up, just label the top-level loop and say `continue <label>`.